src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -614,7 +614,8 @@ private void DecodeWellKnownAttributeAppliedToMethod(ref DecodeWellKnownAttribut
             }
             else if (attribute.IsTargetAttribute(AttributeDescription.UnscopedRefAttribute))
             {
-                if (this.IsValidUnscopedRefAttributeTarget())
+                bool validTarget = this.IsValidUnscopedRefAttributeTarget();
+                if (validTarget && this.UseUpdatedEscapeRules)
```

Comment:
ditto here. Just have a standalone if-block which says if old rules are in use, then report the warning, then continue on to the regular/existing logic.